wallet/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -1593,6 +1597,54 @@ instance BuildableSafeGen WalletSoftwareUpdate where
         updBlockchainVersion
         updScriptVersion
 
+data WalletBalance = WalletBalance
+  { wbBalance  :: V1 (Core.Coin)
+  , wbWalletId :: Text
+  } deriving (Show, Eq, Generic)
+deriveJSON Aeson.defaultOptions ''WalletBalance
+
+instance ToSchema WalletBalance where
+    declareNamedSchema =
+        genericSchemaDroppingPrefix "wb" (\(--^) props -> props
+            & "balance"
+            --^ "the balance of a given public key"
+        )
+
+instance Example WalletBalance
+instance Arbitrary WalletBalance where
+    arbitrary = WalletBalance <$> arbitrary <*> arbitrary
+
+instance Buildable WalletBalance where
+    build (WalletBalance bal walid) = "WalletBalance { ballence = " <> (show bal) <> ", walletid = " <> show walid <> " }"
+
+instance ToSchema EncryptedSecretKey where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "EncryptedSecretKey") $ mempty
+            & type_ .~ SwaggerString
+            & format ?~ "hex|base16"
+
+instance ToJSON EncryptedSecretKey where
+  toJSON _esk = "todo"
+
+mkEncryptedSecretKey :: Text -> Either Text EncryptedSecretKey
+mkEncryptedSecretKey eskhex = join $
+  case Base16.decode eskhex of
+    Left e -> Left e
+    Right bs -> do
+      let
+        esk :: Either Text EncryptedSecretKey
+        esk = decodeFull' bs
+      pure esk
```

Comment:
instead of `pure esk` which then require an extra `join`, what about just going for `Right bs -> decodeFull' bs` ?

Reply:
ah, i think the `do` was messing with things, and adding an extra `Either` ontop of things